src/main/scala/org/constellation/p2p/Download.scala

```diff
@@ -426,12 +412,23 @@ class DownloadProcess[F[_]: Concurrent: Timer: Clock](
       .flatMap(_ => snapshotsProcessor.processSnapshots(hashes))
       .flatTap(_ => dao.metrics.updateMetricAsync("downloadSecondPassComplete", "true"))
 
-  private def finishDownload(snapshot: SnapshotInfo): F[Unit] =
+  private def removeSnapshotInfoFromDisk(snapshotHeight: Long,
+                                         overWritePath: String = dao.snapshotInfoPath.pathAsString) = {
+    val snapshotInfoAtHeight = dao.snapshotInfoPath.collectChildren(_.isDirectory)
+      .filter { file =>
+        val Array(height, hash) = file.name.split('_')
+        height == snapshotHeight.toString
+      }.toList//need to call toList here
+    Sync[F].delay { snapshotInfoAtHeight.foreach(_.delete()) }
+  }
+
+  private def finishDownload(snapshotInfo: SnapshotInfo): F[Unit] =
     for {
-      _ <- setSnapshot(snapshot)
-      _ <- acceptSnapshotCacheData(snapshot)
-      _ <- LiftIO[F].liftIO(dao.snapshotBroadcastService.applyAfterRedownload(snapshot))
-      _ <- storeSnapshotInfo
+      _ <- setSnapshot(snapshotInfo)
+      _ <- acceptSnapshotCacheData(snapshotInfo)
+      _ <- LiftIO[F].liftIO(dao.snapshotBroadcastService.applyAfterRedownload(snapshotInfo))
+      _ <- removeSnapshotInfoFromDisk(snapshotInfo.snapshot.height).map(_ => storeSnapshotInfo)
```

Comment:
Should be `flatMap` probably because with `map` you get `IO[IO[...]]` so `storeSnapshotInfo` won't be triggered. Actually why it is duplicated in line 430 and 431?

Reply:
431 was meant to be deleted. And these explain those missing snapshot infos 💪